// @react95/core/esm/themes/tokens/index.mjs
import { tokens } from "./azureOrange.mjs";
import { tokens as tokens2 } from "./bee.mjs";
import { tokens as tokens3 } from "./blackAndWhite.mjs";
import { tokens as tokens4 } from "./brick.mjs";
import { tokens as tokens5 } from "./candy.mjs";
import { tokens as tokens6 } from "./coldGray.mjs";
import { tokens as tokens7 } from "./counterStrike.mjs";
import { tokens as tokens8 } from "./eggplant.mjs";
import { tokens as tokens9 } from "./highContrast.mjs";
import { tokens as tokens10 } from "./lilac.mjs";
import { tokens as tokens11 } from "./lilacRoseDark.mjs";
import { tokens as tokens12 } from "./maple.mjs";
import { tokens as tokens13 } from "./marine.mjs";
import { tokens as tokens14 } from "./matrix.mjs";
import { tokens as tokens15 } from "./millenium.mjs";
import { tokens as tokens16 } from "./modernDark.mjs";
import { tokens as tokens17 } from "./molecule.mjs";
import { tokens as tokens18 } from "./ninjaTurtles.mjs";
import { tokens as tokens19 } from "./olive.mjs";
import { tokens as tokens20 } from "./pamelaAnderson.mjs";
import { tokens as tokens21 } from "./plum.mjs";
import { tokens as tokens22 } from "./rainyDay.mjs";
import { tokens as tokens23 } from "./rose.mjs";
import { tokens as tokens24 } from "./slate.mjs";
import { tokens as tokens25 } from "./spruce.mjs";
import { tokens as tokens26 } from "./storm.mjs";
import { tokens as tokens27 } from "./theSixtiesUSA.mjs";
import { tokens as tokens28 } from "./tokyoDark.mjs";
import { tokens as tokens29 } from "./tooSexy.mjs";
import { tokens as tokens30 } from "./travel.mjs";
import { tokens as tokens31 } from "./vaporTeal.mjs";
import { tokens as tokens32 } from "./vermillion.mjs";
import { tokens as tokens33 } from "./violetDark.mjs";
import { tokens as tokens34 } from "./water.mjs";
import { tokens as tokens35 } from "./win95.mjs";
export {
  tokens as azureOrange,
  tokens2 as bee,
  tokens3 as blackAndWhite,
  tokens4 as brick,
  tokens5 as candy,
  tokens6 as coldGray,
  tokens7 as counterStrike,
  tokens8 as eggplant,
  tokens9 as highContrast,
  tokens10 as lilac,
  tokens11 as lilacRoseDark,
  tokens12 as maple,
  tokens13 as marine,
  tokens14 as matrix,
  tokens15 as millenium,
  tokens16 as modernDark,
  tokens17 as molecule,
  tokens18 as ninjaTurtles,
  tokens19 as olive,
  tokens20 as pamelaAnderson,
  tokens21 as plum,
  tokens22 as rainyDay,
  tokens23 as rose,
  tokens24 as slate,
  tokens25 as spruce,
  tokens26 as storm,
  tokens27 as theSixtiesUSA,
  tokens28 as tokyoDark,
  tokens29 as tooSexy,
  tokens30 as travel,
  tokens31 as vaporTeal,
  tokens32 as vermillion,
  tokens33 as violetDark,
  tokens34 as water,
  tokens35 as win95
};
