// @react95/core/esm/TitleBar/TitleBar.mjs
import cn from "classnames";
import React from "react";
import close from "./close.svg.mjs";
import help from "./help.svg.mjs";
import maximize from "./maximize.svg.mjs";
import minimize from "./minimize.svg.mjs";
import restore from "./restore.svg.mjs";
import { fixedForwardRef, Frame } from "../Frame/Frame.mjs";
import { button } from "../Button/Button.css.mjs";
import { optionsBox, option, optionImage, titleBarBackground, title } from "./TitleBar.css.mjs";
const defaultOptions = {
  help: {
    alt: "help",
    src: help
  },
  close: {
    alt: "close",
    src: close
  },
  maximize: {
    alt: "maximize",
    src: maximize
  },
  minimize: {
    alt: "minimize",
    src: minimize
  },
  restore: {
    alt: "restore",
    src: restore
  }
};
const OptionsBoxComponent = fixedForwardRef(
  (rest, ref) => /* @__PURE__ */ React.createElement(
    Frame,
    {
      ...rest,
      ref,
      className: cn(optionsBox, rest.className)
    }
  )
);
const OptionsBox = OptionsBoxComponent;
const OptionComponent = fixedForwardRef((rest, ref) => /* @__PURE__ */ React.createElement(
  Frame,
  {
    as: "button",
    ...rest,
    ref,
    className: cn(button, option, rest.className)
  }
));
const Option = OptionComponent;
const DefaultOptionComponent = fixedForwardRef((props, ref) => {
  const { kind, ...rest } = props;
  const optionType = defaultOptions[kind] || defaultOptions.help;
  return /* @__PURE__ */ React.createElement(Option, { ...rest, ref }, /* @__PURE__ */ React.createElement(
    "img",
    {
      src: optionType.src,
      alt: optionType.alt,
      className: optionImage
    }
  ));
});
const DefaultOption = DefaultOptionComponent;
const HelpComponent = fixedForwardRef(
  (props, ref) => {
    return /* @__PURE__ */ React.createElement(DefaultOption, { ...props, kind: "help", ref });
  }
);
const Help = HelpComponent;
const CloseComponent = fixedForwardRef((props, ref) => {
  return /* @__PURE__ */ React.createElement(DefaultOption, { ...props, kind: "close", ref });
});
const Close = CloseComponent;
const MaximizeComponent = fixedForwardRef((props, ref) => {
  return /* @__PURE__ */ React.createElement(DefaultOption, { ...props, kind: "maximize", ref });
});
const Maximize = MaximizeComponent;
const MinimizeComponent = fixedForwardRef((props, ref) => {
  return /* @__PURE__ */ React.createElement(DefaultOption, { ...props, kind: "minimize", ref });
});
const Minimize = MinimizeComponent;
const RestoreComponent = fixedForwardRef((props, ref) => {
  return /* @__PURE__ */ React.createElement(DefaultOption, { ...props, kind: "restore", ref });
});
const Restore = RestoreComponent;
const TitleBarRenderer = fixedForwardRef(({ children, title: title$1 = "UNKNOWN.EXE", icon, active = true, ...rest }, ref) => /* @__PURE__ */ React.createElement(
  Frame,
  {
    ...rest,
    className: cn(titleBarBackground({ active }), rest.className),
    ref
  },
  icon && icon,
  /* @__PURE__ */ React.createElement("div", { className: title }, title$1),
  children
));
const TitleBar = Object.assign(TitleBarRenderer, {
  Option,
  OptionsBox,
  Help,
  Close,
  Maximize,
  Minimize,
  Restore
});
export {
  TitleBar
};
